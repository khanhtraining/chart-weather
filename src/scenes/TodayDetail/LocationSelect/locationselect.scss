.nav-container {
    display: flex;
    justify-content: space-between;
    padding: 20px;
    .nav-burger {
        width: 30px;
        height: 35px;
        & span {
            width: 100%;
            height: 3px;
            background-color: #ffffff;
            display: block;
            margin: 6px 0;
        }
    }

    .nav-location {
        height: 35px;
        text-align: center;
        color: #ffffff;

        .nav-location-dropdown {
            display: flex;
            align-items: center;
            color: rgb(255, 255, 255);
            .icon-arrow {
                display: inline-block;
                width: 15px;
                height: 15px;
                margin-left: 5px;
                background-size: contain;
                background-image: url("../../../assets/arrow_down.svg");
            }
        }
        .dropdown {
            display: none;
            width: 200px;
            left: 0px;
            background-color: rgb(73, 73, 73);
            position: absolute;
            bottom: -90px;
            .option {
                padding: 3px 10px;
            }
            .option:hover {
                background-color: rgba(255,255,255,0.3);
            }
        }
    }

    .nav-notification {
        width: 30px;
        height: 30px;
        background-size: contain;
        background-image: url("../../../assets/bell_notification.svg");
        cursor: pointer;
    }
}
