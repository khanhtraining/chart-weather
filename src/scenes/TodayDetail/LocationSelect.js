import React, { useState } from 'react'
import './locationselect.scss'

export const LocationSelect = (props) => {
    const { onSelect } = props
    const [selectedOption, setSelectedOption] = useState('')
    onSelect(selectedOption)
    return (
        <div>
            <div className='nav-container'>
                <div className='nav-burger'>
                    <span></span>
                    <span></span>
                    <span></span>
                </div>
                <div className='nav-location'>
                    <div className='nav-location-name'>myENV</div>
                    <div className='nav-location-dropdown'>
                        <select className='nav-location-dropdown' name="locations" id="locations" onChange={(e) => setSelectedOption(e.target.value)} >
                            <option value="Singapore">Singapore</option>
                            <option value="Cambridge">Cambridge</option>
                            <option value="Ho Chi Minh City">Ho Chi Minh</option>
                            <option value="Washington D.C.">Washington</option>
                            <option value="Manchester">Manchester</option>
                            <option value="Venezuela">Venezuela</option>
                        </select>
                    </div>
                </div>
                <div className='nav-notification'></div>
            </div>
        </div >
    )
}

export default LocationSelect
