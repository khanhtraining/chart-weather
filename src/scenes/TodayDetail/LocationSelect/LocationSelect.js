import React from 'react'
import './locationselect.scss'

export const LocationSelect = (props) => {
    return (
        <div>
            <div className='nav-container'>
                <div className='nav-burger'>
                    <span></span>
                    <span></span>
                    <span></span>
                </div>
                <div className='nav-location'>
                    <div className='nav-location-name'>myENV</div>
                    <div className='nav-location-dropdown'>
                        <span className='nav-location-dropdown'>
                            {props.todayDeTail.name}
                        </span>
                        <span className='icon-arrow'></span>
                    </div>
                </div>
                <div className='nav-notification'></div>
            </div>
        </div>
    )
}

export default LocationSelect
