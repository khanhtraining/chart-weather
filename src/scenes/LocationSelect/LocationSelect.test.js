import React from 'react'
import LocationSelect from './LocationSelect'
import renderer from 'react-test-renderer'

describe('snapshot testing', () => {
    test('snapshot for Location Select component', () => {
        const renderedComponent = renderer.create(<LocationSelect />).toJSON()
        expect(renderedComponent).toMatchSnapshot()
    })
})
