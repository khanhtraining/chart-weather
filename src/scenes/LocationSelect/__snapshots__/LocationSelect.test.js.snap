// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Location Select component 1`] = `
<div>
  <div
    className="nav-container"
  >
    <div
      className="nav-burger"
    >
      <span />
      <span />
      <span />
    </div>
    <div
      className="nav-location"
    >
      <div
        className="nav-location-name"
      >
        myENV
      </div>
      <div
        className="nav-location-dropdown"
      >
        <select
          className="nav-location-dropdown"
          data-testid="select"
          onChange={[Function]}
        >
          <option
            avalue="Singapore"
          >
            Singapore
          </option>
          <option
            value="Cambridge"
          >
            Cambridge
          </option>
          <option
            value="Ho Chi Minh City"
          >
            Ho Chi Minh
          </option>
          <option
            value="Washington D.C."
          >
            Washington
          </option>
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Venezuela"
          >
            Venezuela
          </option>
        </select>
      </div>
    </div>
    <div
      className="nav-notification"
    />
  </div>
</div>
`;

exports[`snapshot testing snapshot for Location component 1`] = `
<div>
  <div
    className="nav-container"
  >
    <div
      className="nav-burger"
    >
      <span />
      <span />
      <span />
    </div>
    <div
      className="nav-location"
    >
      <div
        className="nav-location-name"
      >
        myENV
      </div>
      <div
        className="nav-location-dropdown"
      >
        <select
          className="nav-location-dropdown"
          data-testid="select"
          onChange={[Function]}
        >
          <option
            value="Singapore"
          >
            Singapore
          </option>
          <option
            value="Cambridge"
          >
            Cambridge
          </option>
          <option
            value="Ho Chi Minh City"
          >
            Ho Chi Minh
          </option>
          <option
            value="Washington D.C."
          >
            Washington
          </option>
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Venezuela"
          >
            Venezuela
          </option>
        </select>
      </div>
    </div>
    <div
      className="nav-notification"
    />
  </div>
</div>
`;
